pkg/persistence/config/loader/config_entry_loader.go

```diff
@@ -229,20 +228,12 @@ func getConfigFromDefinition(
 		parameters[config.ScopeParameter] = scopeParam
 	}
 
-	// if we have an insertAfter field, we should parse it
+	// if we have an insertAfter field, we need to parse the field as a parameter
 	if configType.InsertAfter != nil {
+
 		insertAfterParam, err := parseParameter(fs, context, environment, configId, config.InsertAfterParameter, configType.InsertAfter)
 		if err != nil {
-			return config.Config{}, []error{fmt.Errorf("failed to parse insertAfter: %w", err)}
-		}
-
-		r, isRef := insertAfterParam.(*reference.ReferenceParameter)
-		if !isRef {
-			return config.Config{}, []error{fmt.Errorf("failed to parse insertAfter: cannot use parameter-type %q. Allowed types: %v", insertAfterParam.GetType(), reference.ReferenceParameterType)}
-		}
-
-		if r.Property != "id" {
-			return config.Config{}, []error{fmt.Errorf("failed to parse insertAfter: property field of reference parameter %q must be %q", insertAfterParam, "id")}
+			return config.Config{}, []error{fmt.Errorf("failed to parse insertAfter parameter: %w", err)}
```

Comment:
There is missing coverage on this error, would be great if you could add test case to cover this error.

Reply:
Thanks for pointing out, I  added a test